app/controllers/api/v1/push/subscriptions_controller.rb

```diff
@@ -1,26 +1,31 @@
 # frozen_string_literal: true
 
 class Api::V1::Push::SubscriptionsController < Api::BaseController
+  include Redisable
+  include Lockable
+
   before_action -> { doorkeeper_authorize! :push }
   before_action :require_user!
-  before_action :set_push_subscription
+  before_action :set_push_subscription, only: [:show, :update]
   before_action :check_push_subscription, only: [:show, :update]
 
   def show
     render json: @push_subscription, serializer: REST::WebPushSubscriptionSerializer
   end
 
   def create
-    @push_subscription&.destroy!
-
-    @push_subscription = Web::PushSubscription.create!(
-      endpoint: subscription_params[:endpoint],
-      key_p256dh: subscription_params[:keys][:p256dh],
-      key_auth: subscription_params[:keys][:auth],
-      data: data_params,
-      user_id: current_user.id,
-      access_token_id: doorkeeper_token.id
-    )
+    with_redis_lock("push_subscription:#{current_user.id}") do
+      Web::PushSubscription.where(access_token_id: doorkeeper_token.id).destroy_all
```

Comment:
Ignorable style advice:

- I believe there's a model which the doorkeeper token class includes ... could add an association there to be able to use `doorkeeper_token.web_push_subscriptions....` here.
- Pull out `destroy_previous_web_push_subscriptions` private method and call it from here and from the destroy action below.